Weather/Services/UserLocations/RealmUserLocationsService.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import RealmSwift
+
+
+class RealmWeatherLocation: Object {
+    dynamic var locationId: String = ""
+    dynamic var name: String = ""
+    dynamic var region: String = ""
+    dynamic var country: String = ""
+    var latitude = RealmOptional<Double>()
+    var longitude = RealmOptional<Double>()
+}
+
+
+class RealmUserLocationsService: UserLocationsService {
+    
+    private let realm: Realm?
+    
+    init() {
+        do {
+            self.realm = try Realm()
+        } catch let error {
+            self.realm = nil
+            print(error)
+        }
+    }
+    
+    // MARK: <UserLocationsService>
+    
+    func storeLocation(location: WeatherLocation) {
+        let realmLocation = self.mapLocation(location)
+        
+        do {
+            try self.realm?.write {
+                self.realm?.add(realmLocation)
+            }
+        } catch let error {
+            print(error)
+        }
+    }
+    
+    func allLocations() -> [WeatherLocation]? {
+        return self.realm?.objects(RealmWeatherLocation.self).map({ (realmLocation) -> WeatherLocation in
+            return self.mapRealmLocation(realmLocation)
+        })
+    }
+    
+    func deleteAllLocations() {
+        if let locations = self.realm?.objects(RealmWeatherLocation.self) {
+            self.realm?.delete(locations)
+        }
+    }
+    
+    // MARK: Private
+    
+    private func mapLocation(location: WeatherLocation) -> RealmWeatherLocation {
```

Comment:
I changed it. However I was thinking on moving this logic to a "mapper" object it would be easier to test. What do you think?